lib/contextual_logger.rb

```diff
@@ -5,94 +5,98 @@
+    context_handler = Context::Handler.new(current_context_for_thread.deep_merge(context))
     context_handler.set!
     if block_given?
       yield
     else
-      context_handler
+      context_handler # JEB, is this actually useful? The set! and reset! wouldn't mean anything, so it would just be calling Context::Handler.new, right? -Colin
     end
   ensure
     context_handler.reset! if block_given?
   end
 
   def current_context_for_thread
-    ContextualLogger::Context::Handler.current_context
+    Context::Handler.current_context
   end
 
-  def format_message(severity, timestamp, progname, message, context)
-    message_with_context = message_with_context(context, message, severity, timestamp, progname)
-
-    if @formatter
-      @formatter.call(severity, timestamp, progname, message_with_context)
-    else
-      "#{message_with_context.to_json}\n"
-    end
+  def debug(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::DEBUG, message || yield, extra_context: extra_context)
   end
 
-  def debug(progname = nil, **extra_context, &block)
-    add(Logger::Severity::DEBUG, nil, progname, extra_context, &block)
+  def info(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::INFO, message || yield, extra_context: extra_context)
   end
 
-  def info(progname = nil, **extra_context, &block)
-    add(Logger::Severity::INFO, nil, progname, extra_context, &block)
+  def warn(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::WARN, message || yield, extra_context: extra_context)
   end
 
-  def warn(progname = nil, **extra_context, &block)
-    add(Logger::Severity::WARN, nil, progname, extra_context, &block)
+  def error(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::ERROR, message || yield, extra_context: extra_context)
   end
 
-  def error(progname = nil, **extra_context, &block)
-    add(Logger::Severity::ERROR, nil, progname, extra_context, &block)
+  def fatal(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::FATAL, message || yield, extra_context: extra_context)
   end
 
-  def fatal(progname = nil, **extra_context, &block)
-    add(Logger::Severity::FATAL, nil, progname, extra_context, &block)
+  def unknown(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::UNKNOWN, message || yield, extra_context: extra_context)
   end
 
-  def unknown(progname = nil, **extra_context, &block)
-    add(Logger::Severity::UNKNOWN, nil, progname, extra_context, &block)
+  def log_level_enabled?(severity)
+    severity >= @level
   end
 
-  def add(severity, message = nil, progname = nil, extra_context = nil)
-    severity ||= UNKNOWN
-    if @logdev.nil? || (severity < @level)
-      return true
+  def add_if_enabled(severity, message, extra_context:)
+    if log_level_enabled?(severity)
+      add(severity, message: message, progname: @progname, extra_context: extra_context)
```

Comment:
But I did try to preserve the original `add` signature in case there's code somewhere that directly calls it.

Reply:
That's good, I think I remember that ActiveSupport::TaggedLogging will call `add` on their own instead of flowing through the rest of the stack.  One of the reasons why this contract was left alone for the most part originally.